python/dataset.cpp

```diff
@@ -253,22 +254,27 @@ void init_dataset(py::module &m) {
   py::class_<DataArray> dataArray(m, "DataArray", R"(
     Named variable with associated coords, masks, and attributes.)");
   dataArray.def(py::init<const DataArrayConstView &>());
-  dataArray.def(
-      py::init([](VariableConstView data,
-                  std::map<Dim, VariableConstView> coords,
-                  std::map<std::string, VariableConstView> masks,
-                  std::map<Dim, VariableConstView> unaligned_coords,
-                  const std::string &name) {
-        return DataArray{Variable{data}, coords, masks, unaligned_coords, name};
-      }),
-      py::arg("data") = Variable{},
-      py::arg("coords") = std::map<Dim, VariableConstView>{},
-      py::arg("masks") = std::map<std::string, VariableConstView>{},
-      py::arg("unaligned_coords") = std::map<Dim, VariableConstView>{},
-      py::arg("name") = std::string{});
+  dataArray
+      .def(py::init([](VariableConstView data,
+                       std::map<Dim, VariableConstView> coords,
+                       std::map<std::string, VariableConstView> masks,
+                       std::map<Dim, VariableConstView> unaligned_coords,
+                       const std::string &name) {
+             return DataArray{Variable{data}, coords, masks, unaligned_coords,
+                              name};
+           }),
+           py::arg("data") = Variable{},
+           py::arg("coords") = std::map<Dim, VariableConstView>{},
+           py::arg("masks") = std::map<std::string, VariableConstView>{},
+           py::arg("unaligned_coords") = std::map<Dim, VariableConstView>{},
+           py::arg("name") = std::string{})
+      .def("__sizeof__",
+           [](const DataArrayConstView &array) { return size_of(array, true); });
 
   py::class_<DataArrayConstView>(m, "DataArrayConstView")
-      .def(py::init<const DataArray &>());
+      .def(py::init<const DataArray &>())
+      .def("__sizeof__",
+           [](const DataArrayConstView &array) { return size_of(array, false); });
```

Comment:
Should be `true`?

Reply:
Yes probably should be. My reasoning here was that if you access a dataarray contained in a dataset you don't want to include the coordinate size but if you slice a data array and produce a view you do. Having thought about it some more however for views we are returning the size they would have in memory.